src/models/legend.js

```diff
@@ -90,13 +91,25 @@ nv.models.legend = function() {
                 .attr('dx', '8');
 
             var seriesText = series.select('text.nv-legend-text');
+            var tooltip = nv.models.tooltip();
 
             series
                 .on('mouseover', function(d,i) {
                     dispatch.legendMouseover(d,i);  //TODO: Make consistent with other event objects
+                    if (getKey(d).length > maxKeyLength) {
+                        d['series'] = {
+                            key: getKey(d)
+                        };
+                        tooltip.duration(0).headerEnabled(false);
+                        tooltip.data(d).hidden(false);
+                        tooltip.data(d).colorDivEnabled(false);
+                        tooltip.data(d).valueEnabled(false);
```

Comment:
Just realized, if the goal is to just show a tooltip with the full legend text, why not use a title element? http://stackoverflow.com/questions/10643426/how-to-add-a-tooltip-to-an-svg-graphic